aredis/pool.py

```diff
@@ -193,20 +194,27 @@ def _checkpid(self):
                 self.disconnect()
                 self.reset()
 
-    def get_connection(self, *args, **kwargs):
+    async def get_connection(self, *args, **kwargs):
         "Get a connection from the pool"
         self._checkpid()
         try:
             connection = self._available_connections.pop()
         except IndexError:
-            connection = self.make_connection()
+            connection = await self.make_connection()
         self._in_use_connections.add(connection)
         return connection
 
-    def make_connection(self):
+    async def make_connection(self):
         "Create a new connection"
-        if self._created_connections >= self.max_connections:
+        wait_for_connection = self.wait_for_connection
+        while self._created_connections >= self.max_connections:
+            if wait_for_connection > 0:
+                wait_for_connection -= 1
+                await asyncio.sleep(1)
```

Comment:
`sleep` for one second each loop seems not so accurate. If user just want to wait the connection for 0.5 seconds, code here will make it one second. I think a retry after `wait_for_connection` time is better than retry in each loop.

Reply:
Hmm, maybe this should be something like `await asyncio.sleep(min(wait_for_connection // 10, 1))`?

Reply:
I think it is better to let user decide how long client should wait and how many times it should retry. Since user should always make a try-catch in case error(not only `too many connections`) during interaction with redis, i prefer to add single sleep here and retry just once, raise if still failed to acquire a connection.